plugins/modules/source_control/gitlab/gitlab_service.py

```diff
@@ -0,0 +1,516 @@
+           "redmine": {"new_issue_url": {"required": 1, "type": "str"}, "project_url": {"required": 1, "type": "str"},
+                       "issues_url": {"required": 1, "type": "str"}},
+           "slack": {"webhook": {"required": 1, "type": "str", "no_log": True}, "username": {"type": "str"}, "channel": {"type": "str"},
+                     "branches_to_be_notified": {"type": "str"}, "notify_only_broken_pipelines": {"type": "bool"}, "push_channel": {"type": "str"},
+                     "issue_channel": {"type": "str"}, "confidential_issue_channel": {"type": "str"}, "merge_request_channel": {"type": "str"},
+                     "note_channel": {"type": "str"}, "tag_push_channel": {"type": "str"}, "pipeline_channel": {"type": "str"},
+                     "wiki_page_channel": {"type": "str"}, "push_events": {"type": "bool"}, "issues_events": {"type": "bool"},
+                     "confidential_issues_events": {"type": "bool"}, "merge_requests_events": {"type": "bool"}, "note_events": {"type": "bool"},
+                     "confidential_note_events": {"type": "bool"}, "tag_push_events": {"type": "bool"}, "pipeline_events": {"type": "bool"},
+                     "wiki_page_events": {"type": "bool"}},
+           "slack-slash-commands": {"token": {"required": 1, "type": "str", "no_log": True}},
+           "teamcity": {"teamcity_url": {"required": 1, "type": "str"}, "build_type": {"required": 1, "type": "str"},
+                        "username": {"required": 1, "type": "str"}, "password": {"required": 1, "type": "str", "no_log": True}},
+           "unify-circuit": {"webhook": {"required": 1, "type": "str", "no_log": True}, "push_events": {"type": "bool"},
+                             "issues_events": {"type": "bool"}, "confidential_issues_events": {"type": "bool"}, "merge_requests_events": {"type": "bool"},
+                             "note_events": {"type": "bool"}, "confidential_note_events": {"type": "bool"}, "tag_push_events": {"type": "bool"},
+                             "pipeline_events": {"type": "bool"}, "wiki_page_events": {"type": "bool"}},
+           "webex-teams": {"webhook": {"required": 1, "type": "str", "no_log": True}, "push_events": {"type": "bool"},
+                           "issues_events": {"type": "bool"}, "confidential_issues_events": {"type": "bool"}, "merge_requests_events": {"type": "bool"},
+                           "note_events": {"type": "bool"}, "confidential_note_events": {"type": "bool"}, "tag_push_events": {"type": "bool"},
+                           "pipeline_events": {"type": "bool"}, "wiki_page_events": {"type": "bool"}},
+           "youtrack": {"project_url": {"required": 1, "type": "str"}, "issues_url": {"required": 1, "type": "str"}}}
+
+
+def init_definitions():
+    for params in SRV_DEF.values():
+        for name, definition in params.items():
+            if 'type' in definition:
+                definition['type'] = bool if definition['type'] == 'bool' else \
```

Comment:
What is this for? `'type'` in argspec should be a string. (A callable is only OK for types that are not directly supported by Ansible.)

Reply:
Since I generate better definition straight from Ruby, these can be removed. (They may also date back from older Ansible versions).
=> Removed the whole `init_definition()` stuff.